### DISCUSSION TITLE
Altitude of native HRRR levels?

### DISCUSSION THREAD

**Discussion by nicholasbalasus** [NOAA-GSL/GSL-Regional-Model-Forum#43] [Q&A] [answered]
Hello - I am interested in using the average winds within the boundary layer with HRRR outputs. 

I am using the native level files.
```
grib_file = "hrrr.t00z.wrfnatf00.grib2"
```

I seem to be able to get the boundary layer height (in meters) like this:
```
with xr.open_dataset(grib_file, engine="cfgrib",\
backend_kwargs={'filter_by_keys': {'stepType': 'instant',\
'typeOfLevel':'surface'}}) as ds:
    
    blh = ds["blh"].values
```

I can get the u and v winds on the pressure grid like this:
```
with xr.open_dataset(grib_file, engine="cfgrib",\
backend_kwargs={'filter_by_keys': {'typeOfLevel':'hybrid'}})\
as ds:
    
    u = ds["u"].values
    v = ds["v"].values
```

I'd like to average all of the u and v winds if their level is below the boundary layer height. I can't seem to find a variable for the height of the hybrid layers that I could compare to the boundary layer height. I suspect I am just missing the variable somewhere (or maybe I can just convert the geopotential height variable?)

Tagging @blaylockbk as they seem to be an expert on this!

Thanks,
Nick

**Comment by EricJames-NOAA** [comment, marked as answer]
@nicholasbalasus thanks for the question and sorry for the very late reply.  I think you should be able to subtract the geopotential height on the native model levels from the geopotential height at the surface to get a geopotential height above ground level.  Then you could convert this to a geometric height for comparison with the PBL height.
